src/models/booking.cr

```diff
@@ -69,6 +115,17 @@ class Booking
     booking_model.approver_email = booking_model.approver_email if booking_model.approver_email_column.defined?
     booking_model.email_digest = booking_model.user_email.digest
     booking_model.booked_by_email_digest = booking_model.booked_by_email.digest
+    booking_model.booked_from = booking_model.utm_source if !booking_model.booked_from_column.defined?
+    booking_model.history = booking_model.current_history
```

Comment:
Should I raise an error in #before(:save) if the current state would be Unknown?

Reply:
Yeah validation is needed for a field like this, keep in mind though there will be bookings made previous to this change so you will have to add a default value for all those previous bookings, probably in the migration

Reply:
I would log an error, with some metadata related to the booking, then save as usual.
This will allow us to investigate while the booking is still persisted.

Reply:
```cr
Log.warn { "Booking is in an Unknown state" } if booking_model.current_state.unknown?
```

Reply:
I would say this is an `error`, as we have failed to cover all the states a booking could be in.
Additionally, there's no way to know which booking is in the `Unknown` state from this log.
It would be good to add some state related to the booking such as the check-in time etc etc, i.e. all the metadata used to calculate a booking's state.

In fact, this might be better done in the actual `current_state` method

Reply:
ok